### ISSUE TITLE
When receiving new data, which is different than previously rendered, table errors with emitter undefined.

### ISSUE THREAD

**Issue by stevelacerda7** [AllenFang/react-bootstrap-table#1642]
Similar to [#1418](https://github.com/AllenFang/react-bootstrap-table/issues/1418). I have a table that's being redrawn on a time interval, as data changes, so the columns and data may change as new data comes in. The problem is that the table doesn't want to cooperate when the data structure changes (new columns are added). I tried replace=true on the BootstrapTable, which only seems to work if I use:

```
setState({ rows: [], columns: [] })
setState({ rows: newData, columns: newData: []);
```

Otherwise, I receive an error:

`bundle.js:223729 Uncaught TypeError: Cannot read property 'emitter' of undefined`

The problem is, that I then get two renders, and I can visually see the render of the empty dataset. Is there anyway to handle the new column or removal of an old column gracefully?

**Comment by AllenFang** [comment]
sorry for lately reply. some question I need to confirm:
1. the columns is totally different with previous?
2. what column configure the filter

You may describe the differentiation for your columns . thanks

**Comment by stevelacerda7** [comment]
1) Yes, columns are totally different. They change depending on the object the user selects. Thus, the columns must all be re-rendered and the old data tossed.
2) Not sure what you mean by number 2. Please clarify.

As I said previously, I was able to clear the headers using setState({ rows: [], columns: [] }). This is working for me, but was not intuitive and took me a few minutes to understand and find a workaround.

**Comment by AllenFang** [comment]
Fixed on `v4.0.6`, let me know if the problem still remain, thanks
